Frontend/src/app/details/details.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RecipeService } from '../recipe.service';
+import { Recipe } from '../RecipeInterface';
+
+@Component({
+  selector: 'app-details',
+  templateUrl: './details.component.html',
+  styleUrls: ['./details.component.css']
+})
+export class DetailsComponent implements OnInit {
+
+  id: string;
+  data: Recipe = this.recipeService.defaultRecipe();
+
+  constructor(
+    private route: ActivatedRoute,
+    private recipeService: RecipeService,
+    private router: Router
+  ) { }
+
+  ngOnInit() {
+    this.route.paramMap.subscribe(params => this.id = params.get('id'));
+    this.recipeService.getOneRecipe(this.id).subscribe(
+      (value: any) => { this.data = value; },
```

Comment:
This shouldn't be any
```suggestion
      (value: Recipe) => { this.data = value; },
```